### PR TITLE
feat: add api 'get_full_schema_type' to return the schema ty with third party

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

Part of #891 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm/api/src/service/capi.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/spec/gpyrpc/gpyrpc.proto [hunk 1]
```diff
@@ -213,6 +214,11 @@ message OverrideFile_Result {
 	bool result = 1;
 }
 
+message GetFullSchemaType_Args {
+	ExecProgram_Args exec_options = 1;
```
Comment (Peefy) [inline kclvm/spec/gpyrpc/gpyrpc.proto:218]: s/exec_options/exec_args

File: kclvm/query/src/query.rs [hunk 2]
```diff
@@ -106,30 +117,58 @@ pub fn get_schema_type(
...
-fn resolve_file(file: &str, code: Option<&str>) -> Result<Rc<RefCell<Scope>>> {
+#[derive(Debug, Clone, Default)]
+pub struct CompilationOptions {
+    pub k_files: Vec<String>,
+    pub loader_opts: Option<LoadProgramOptions>,
+    pub resolve_opts: Options,
+    pub get_schema_opts: GetSchemaOption,
+}
+
+pub fn get_full_schema_type(
```
Comment (Peefy) [inline kclvm/query/src/query.rs:128]: Could U pls add more documents for public functions and structures?

File: kclvm/api/src/service/jsonrpc.rs [hunk 3]
```diff
@@ -91,6 +91,14 @@ fn register_kclvm_service(io: &mut IoHandler) {
         };
         futures::future::ready(catch!(kclvm_service_impl, args, get_schema_type))
     });
+    io.add_method("KclvmService.GetFullSchemaType", |params: Params| {
```
Comment (Peefy) [inline kclvm/api/src/service/jsonrpc.rs:94]: Can you help add the missing API list? Including Test, RenameCode, RenameFile, etc?

Besides, add function list in the `register_builtin_service` function.

File: kclvm/query/Cargo.toml [hunk 4]
```diff
@@ -16,6 +16,7 @@ kclvm-ast-pretty = {path = "../ast_pretty"}
 kclvm-parser = {path = "../parser"}
 kclvm-sema = {path = "../sema"}
 kclvm-error = {path = "../error"}
+maplit = "1.0.2"
```
Comment (Peefy) [inline kclvm/query/Cargo.toml:19]: Does this crate `maplit` have been used?

File: kclvm/api/src/service/capi.rs [hunk 5]
```diff
@@ -180,6 +181,18 @@ pub(crate) fn get_schema_type(serv: *mut kclvm_service, args: *const c_char) ->
     call!(serv, args, GetSchemaTypeArgs, get_schema_type)
 }
 
+/// Get full schema types from a kcl file or code.
+///
+/// # Parameters
+/// `exec_options`: [Option<ExecProgramArgs>]
```
Comment (Peefy) [inline kclvm/api/src/service/capi.rs:187]: s/exec_options/exec_args

File: kclvm/api/src/service/service_impl.rs [hunk 6]
```diff
@@ -201,6 +205,63 @@ impl KclvmServiceImpl {
...
+    ///         CmdExternalPkgSpec{
+    ///             pkg_name:"bbb".to_string(),
+    ///             pkg_path: work_dir_parent.join("bbb").canonicalize().unwrap().display().to_string()
+    ///         }
+    ///     ],
+    ///     ..Default::default()
+    /// };
+    ///
+    /// let result = serv.get_full_schema_type(&GetFullSchemaTypeArgs {
+    ///     exec_options: Some(args),
```
Comment (Peefy) [inline kclvm/api/src/service/service_impl.rs:234]: s/exec_options/exec_args

File: kclvm/query/src/query.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/query/src/query.rs:159]: fixed

File: kclvm/api/src/service/service_impl.rs (hunk 6 above)
Comment (zong-zhe) [reply kclvm/api/src/service/service_impl.rs]: fixed

File: kclvm/api/src/service/capi.rs (hunk 5 above)
Comment (zong-zhe) [reply kclvm/api/src/service/capi.rs]: fixed

File: kclvm/query/Cargo.toml (hunk 4 above)
Comment (zong-zhe) [reply kclvm/query/Cargo.toml:19]: fixed

File: kclvm/query/src/query.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/query/src/query.rs:159]: fixed

File: kclvm/spec/gpyrpc/gpyrpc.proto (hunk 1 above)
Comment (zong-zhe) [reply kclvm/spec/gpyrpc/gpyrpc.proto]: fixed.

File: kclvm/api/src/service/jsonrpc.rs (hunk 3 above)
Comment (zong-zhe) [reply kclvm/api/src/service/jsonrpc.rs:94]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**